tasks.py

```diff
@@ -1,28 +1,70 @@
 from invoke import task
+from invoke.exceptions import Failure
+from invoke.tasks import call
 
 
+def run_or_fail(c, cmd, fail_message="Error", fail_callback=None):
+    try:
+        cmd(c)
+    except Failure as ex:
+        print(fail_message)
+        if fail_callback:
+            fail_callback(c)
+        exit()
+
 @task()
+def tools_check(c, tools=[]):
+    missing_tools = []
+    for tool in tools:
+        if c.run(f"which {tool}", hide=True, warn=True).exited != 0:
+            missing_tools.append(f"{tool}")
+
+    if len(missing_tools) > 0:
+        print("These tools are required : {}".format(", ".join(missing_tools)))
+        print("Please install them")
+        exit()
+
+
+@task(pre=[call(tools_check, ["yarn"])])
 def setup_frontend(c):
     print("Running frontend setup...")
 
     with c.cd("frontend"):
         c.run("yarn")
 
 
-@task()
+@task(pre=[call(tools_check, ["docker", "docker-compose"])])
+def setup_db(c):
+    print("Running DB setup...")
+
+    with c.cd("backend"):
+        run_or_fail(c, lambda x: x.run("ls .env.sample", hide=False),
+                    fail_message="Are you in project root dir?")
+        run_or_fail(c, lambda x: x.run("ls .env", hide=False),
+                    fail_message="ERROR! Have you copied backend/.env.sample "
+                                 "in backend/.env and set your .env vars?")
```

Comment:
Maybe it's useful to some util for preconditions? 

So this code can be:

```
[...]

check_env_precondition()
```

in case we need to reuse this same check somewhere else